libtenzir/builtins/formats/cef.cpp

```diff
@@ -40,25 +40,48 @@ namespace tenzir::plugins::cef {
 namespace {
 
 /// Unescapes CEF string data containing \r, \n, \\, and \=.
-std::string unescape(std::string_view value) {
-  std::string result;
-  result.reserve(value.size());
-  for (auto i = 0u; i < value.size(); ++i) {
-    if (value[i] != '\\') {
-      result += value[i];
-    } else if (i + 1 < value.size()) {
-      auto next = value[i + 1];
-      switch (next) {
-        default:
-          result += next;
-          break;
-        case 'r':
-        case 'n':
-          result += '\n';
-          break;
+auto unescape(std::string_view::iterator begin, std::string_view::iterator end,
+              std::back_insert_iterator<std::string> out)
+  -> std::string_view::iterator {
+  TENZIR_ASSERT_EXPENSIVE(*std::prev(begin) == '\\');
+  TENZIR_ASSERT_EXPENSIVE(begin < end);
+  switch (*begin) {
+    case 'n': {
+      out = '\n';
+      return ++begin;
+    }
+    case 'r': {
+      out = '\n';
+      return ++begin;
+    }
+    case '=': {
+      out = '=';
+      return ++begin;
+    }
+    case '\\': {
+      out = '\\';
+      return ++begin;
+    }
+    default: {
+      return begin;
+    }
+  }
+  TENZIR_UNREACHABLE();
+  return begin;
```

Comment:
Unreachable followed by return, and also `default: return` but then also something afterwards.